js-react/README.md

```diff
@@ -0,0 +1,340 @@
+
+```js
+function Heading() {
+  const title = 'Root finding web application';
+  return <h1>{title}</h1>
+}
+
+ReactDOM.render(
+  <Heading/>,
+  document.getElementById('container')
+);
+```
+
+When the page is rendered, the generated HTML code will be like:
+
+```html
+<h1>Root finding web application</h1>;
+```
+
+## Adding the  web form
+
+The web application in our example should have a form with ``tolerance`` and ``initial_guess`` input fields, as well as a submit button. The form in JSX can be written in the following way:
+
+```js
+<form onSubmit={handleSubmit}>
+  <label>
+    Tolerance:
+    <input name="tolerance" type="number" value={tolerance} onChange={onToleranceChange}/>
+  </label>
+  <label>
+    Initial guess:
+    <input name="initial_guess" type="number" value={initial_guess} onChange={onGuessChange}/>
+  </label>
+  <input type="submit" value="Submit" />
+</form>
+```
+
+The form tag has a `onSubmit` property, which is set to a function (`handleSubmit`) that will handle the form
+submission. The input tag has a `value` property to set the variable (`tolerance` and `initial_guess`) and it also has `onChange`
+property to set the function (`onToleranceChange` and `onGuessChange`) which will be triggered when the user changes the
+value.
+
+Let's implement the `value` and `onChange` for the `tolerance` input.
+To store the value we will use the [React useState hook](https://reactjs.org/docs/hooks-state.html).
+
+```js
+const [tolerance, setTolerance] = React.useState(0.001);
+```
+
+The argument of the `useState` function is the initial value. The `tolerance` variable contains the current value for
+tolerance and `setTolerance` is a function to set tolerance to a new value.
+
+The input tag in the form will call the `onChange` function with a event object. We need to extract the user input from
+the event and pass it to `setTolerance`. The value should be a number, so we use `Number()` to cast the string from the
+event to a number.
+
+```js
+function onToleranceChange(event) {
+  setTolerance(Number(event.target.value));
+}
+```
+
+We will follow the same steps for the `initial_guess` input as well.
+
+```js
+const [initial_guess, setGuess] = React.useState(-20);
+
+function onGuessChange(event) {
+  setGuess(Number(event.target.value));
+}
+```
+
+We are ready to implement the `handleSubmit` function which will process the form data. The function will get, similar
+to the `onChange` of the input tag, an event object. Normally when you submit a form the form fields will be send to the
+server, but we want to perform the calculation in the browser so we have to disable the default action with.
+
+```jsx
+event.preventDefault();
+```
+
+Like we did in the previous post we have to construct a web worker.
+
+```jsx
+const worker = new Worker('worker.js');
+```
+
+The `worker.js` is the same as in the previous post so we re-use it by
+
+```jsx
+cd react && ln -s ../webassembly/worker.js . && cd -
+```
```

Comment:
The worker will not have a sleep in it, so it will not be an exact copy from ../js-webapp-async/worker.js

Once code is included as separate files for this post we can link to it, instead of asking user to  symlink.

Reply:
will adress in next pr